src/main/java/io/jenkins/plugins/coverage/CoveragePublisher.java

```diff
@@ -40,11 +41,11 @@
     private boolean failUnstable;
     private boolean failNoReports;
 
-    private String autoDetectPath;
+    private Detector reportDetector;
 
     @DataBoundConstructor
-    public CoveragePublisher(String autoDetectPath) {
-        this.autoDetectPath = autoDetectPath;
+    public CoveragePublisher(Detector reportDetector) {
```

Comment:
Is it really required? You already have DataBoundSetter

Reply:
I think `detector` is a necessary parameter, so I add it to the constructor.